src/packages/engine-core/src/NAPIEngine.ts

```diff
@@ -0,0 +1,547 @@
+import Debug from '@prisma/debug'
+import { getEnginesPath } from '@prisma/engines'
+import { DMMF } from '@prisma/generator-helper'
+import { getNapiName, getPlatform, Platform } from '@prisma/get-platform'
+import chalk from 'chalk'
+import EventEmitter from 'events'
+import fs from 'fs'
+import path from 'path'
+import type {
+  DatasourceOverwrite,
+  Engine,
+  EngineConfig,
+  EngineEventType,
+  GetConfigResult,
+} from './Engine'
+import {
+  PrismaClientInitializationError,
+  PrismaClientKnownRequestError,
+  PrismaClientUnknownRequestError,
+  RequestError,
+} from './errors'
+import { printGeneratorConfig } from './printGeneratorConfig'
+import { fixBinaryTargets } from './util'
+const debug = Debug('prisma:client:napi')
+
+const MAX_REQUEST_RETRIES = process.env.PRISMA_CLIENT_NO_RETRY ? 1 : 2
+type QueryEngineLogLevel = 'trace' | 'debug' | 'info' | 'warn' | 'error' | 'off'
+type QueryEngineEvent = QueryEngineLogEvent | QueryEngineQueryEvent
+type QueryEngineConfig = {
+  datamodel: string
+  datasourceOverrides?: Record<string, string>
+  logLevel: QueryEngineLogLevel
+}
+type QueryEngineLogEvent = {
+  level: string
+  module_path: string
+  message: string
+}
+type QueryEngineQueryEvent = {
+  level: 'info'
+  module_path: string
+  query: string
+  item_type: 'query'
+  params: string
+  duration_ms: string
+  result: string
+}
+
+export interface QueryEngineConstructor {
+  new (
+    config: QueryEngineConfig,
+    logger: (err: string, log: string) => void,
+  ): QueryEngine
+}
+
+type ConnectArgs = {
+  enableRawQueries: boolean
+}
+
+export type QueryEngine = {
+  connect(connectArgs: ConnectArgs): Promise<void>
+  disconnect(): Promise<void>
+  getConfig(): Promise<GetConfigResult>
+  dmmf(): Promise<DMMF.Document>
+  query(request: any): Promise<string>
+  sdlSchema(): Promise<string>
+  serverInfo(): Promise<string>
+  nextLogEvent(): Promise<string>
+}
+
+type ServerInfo = {
+  commit: string
+  version: string
+  primaryConnector: string
+}
+
+type SyncRustError = {
+  is_panic: boolean
+  message: string
+  meta: {
+    full_error: string
+  }
+  error_code: string
+}
+
+type RustRequestError = {
+  is_panic: boolean
+  message: string
+  backtrace: string
+}
+function isQueryEvent(event: QueryEngineEvent): event is QueryEngineQueryEvent {
+  return event.level === 'info' && event['item_type'] === 'query'
+}
+
+const knownPlatforms: Platform[] = [
```

Comment:
Maybe we could import from https://github.com/prisma/prisma/blob/6120a66c4eb2242e5821fac75a871442b0294671/src/packages/get-platform/src/platforms.ts#L19 to not duplicate>

Reply:
It's also duplicated in NodeEngine